packages/vm/lib/runTx.ts

```diff
@@ -383,3 +407,64 @@ function txLogsBloom(logs?: any[]): Bloom {
   }
   return bloom
 }
+
+/**
+ * Returns the tx receipt.
+ */
+export async function generateTxReceipt(
+  this: VM,
+  tx: TypedTransaction,
+  txResult: RunTxResult,
+  blockGasUsed: BN,
+  debug?: Function
```

Comment:
Why this param?

Reply:
If a `debug` instance is passed in, it will do `debug(log)` for the tx receipt debug log message.

Reply:
Ah, let's please not start now to add `debug` flags to every single method, this will get messy soon and should be handled more carefully, especially since this is not specially urgent or something.

Have opened a corresponding issue on the overall eventual problem set: #1186

Reply:
@holgerd77 i agree, this was just a minimally invasive way for me to keep the debug log without too much additional complexity. do you have a suggestion on an alternative? i thought it'd be mostly harmless since it's optional.

Reply:
I am not completely getting the context, what does "keep" mean here? Can't this just run in the normal `debug` context we are using all over the library?

Reply:
@holgerd77 take a look at this runTx `generateTxReceipt` function, there is a debug log message assembled inside with some of the tx receipt information.

Reply:
I would still prefer here to just use the `debug` object generated in the `runTx.ts` module and not pass things over: just tested this by removing the parameter from the function signature and then run `DEBUG=vm:* npx tape -r ts-node/register --stack-size=1500 ./tests/api/buildBlock.spec.ts` and this works fine:

![image](https://user-images.githubusercontent.com/931137/114010921-ede08d00-9864-11eb-986e-81367e0a2991.png)

(or am I missing a usage context where this is not available?)

Reply:
ohh I see what you mean now, sorry it took me a second, I didn't realize `const debug` was created in the global scope, so that makes it easy it can just be used, I will remove the func param 🙂

Reply:
fixed in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1185/commits/8cb3c3ca0addcb3c40a33e4504a6c8ea13e601ea